src/ui/ons/ONSSelect.svelte

```diff
@@ -1,15 +1,41 @@
 <script>
-  export let label;
-  export let id;
-  export let name;
+  import ONSError from "./partials/ONSError.svelte";
+  export let label, id, name, renderError, selectValue, onClick;
+  export let errorText = "Select [whatever topic the drop-down is asking for]";
   export let options = [];
+
+  function disablesSelectedOption(options) {
+    let selectOption = document.getElementById(id);
+    if (selectOption.options[selectOption.options.selectedIndex].selected) {
+      options.find((item) => {
+        if (item.value === selectOption.options[selectOption.options.selectedIndex].value) {
+          item.disabled = true;
+        } else {
+          item.disabled = false;
+        }
+      });
+    }
+  }
 </script>
 
-<div class="ons-field">
-  <label class="ons-label " for={name}>{label}</label>
-  <select {id} {name} class="ons-input ons-input--select">
-    {#each options as option}
-      <option value={option.value} selected={option.selected} disabled={option.disabled}>{option.label}</option>
-    {/each}
-  </select>
-</div>
+<ONSError {errorText} {id} {renderError}>
+  <div class="ons-field">
+    <label class="ons-label " for={id}>{label}</label>
+    <select
+      on:change={() => {
+        disablesSelectedOption(options);
+      }}
```

Comment:
Same as above.

Reply:
So basically you are saying to completely remove the on:change event ?

Reply:
Yep, I'd remove it for now and tackle these events when we introduce some validation.

Reply:
The angle on validation would depend on whether we cater for Census Atlas only (= simple, minimal, per page/screen validation), or whether we aim for Svelte Design System (= introducing a validation schema like [yup](https://github.com/jquense/yup) or similar). This would be applicable to all form elements, not just Select. We would need to refactor our form fields to accept errorText as well.

Reply:
Done.